sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -638,7 +641,44 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
       parameters = Map(
         "fieldName" -> "c2",
         "expressionStr" -> "UCASE(struct1.a)",
-        "reason" -> "generation expression cannot contain non-default collated string type"))
+        "reason" ->
+          "generation expression cannot contain non-binary orderable collated string type"))
+  }
+
+  test("SPARK-47431: Default collation set to UNICODE, literal test") {
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> "UNICODE") {
+      checkAnswer(sql(s"SELECT collation('aa')"), Seq(Row("UNICODE")))
+    }
+  }
+
+  test("SPARK-47431: Default collation set to UNICODE, column type test") {
+    withSQLConf(SqlApiConf.DEFAULT_COLLATION -> "UNICODE") {
+      sql(
+        s"""
+           |CREATE TABLE t(c1 STRING)
+           |USING PARQUET
+           |""".stripMargin)
+      sql("INSERT INTO t VALUES ('a')")
+      checkAnswer(sql("SELECT collation(c1) FROM t"), Seq(Row("UNICODE")))
+    }
+  }
+
+  test("SPARK-47431: Create table with UTF8_BINARY, make sure collation persists on read") {
+    val tableName = "t"
+    withTable(tableName) {
+      withSQLConf(SqlApiConf.DEFAULT_COLLATION -> "UTF8_BINARY") {
+        sql(
+          s"""
+             |CREATE TABLE $tableName(c1 STRING)
+             |USING PARQUET
+             |""".stripMargin)
+        sql(s"INSERT INTO $tableName VALUES ('a')")
+        checkAnswer(sql(s"SELECT collation(c1) FROM $tableName"), Seq(Row("UTF8_BINARY")))
+        checkAnswer(sql("SET spark.sql.session.collation.default=UNICODE"),
```

Comment:
we should have two `withSQLConf` blocks, one for CREATE TABLE, one for the final SELECT